From 1bb509974e9100ea87ea44c59d73f622c6e11b9a Mon Sep 17 00:00:00 2001
From: Xiaofeng Wei <xiaofeng.wei@nxp.com>
Date: Mon, 17 Feb 2025 16:39:36 +0800
Subject: [PATCH 24/30] arm64: dts: add dual BOE EV121WXM-N10-1850 LVDS panel
 support for i.MX8MP FRDM

Add dual BOE EV121WXM-N10-1850 LVDS panel support for i.MX8MP FRDM board.

Signed-off-by: Xiaofeng Wei <xiaofeng.wei@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  2 +-
 .../imx8mp-frdm-boe-wxga-lvds0-panel.dts      | 81 +++++++++++++++++++
 .../imx8mp-frdm-boe-wxga-lvds1-panel.dts      | 81 +++++++++++++++++++
 arch/arm64/boot/dts/freescale/imx8mp.dtsi     |  6 +-
 4 files changed, 168 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wxga-lvds0-panel.dts
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wxga-lvds1-panel.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index bd3ea6588619..ecb0e26ff2c5 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -141,7 +141,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mp-navq.dtb imx8mp-navq-ov5640-ov5645.dtb imx8mp-n
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-ddr4-evk.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-evk-ndm.dtb
 
-dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm.dtb imx8mp-frdm-os08a20.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm.dtb imx8mp-frdm-os08a20.dtb imx8mp-frdm-boe-wxga-lvds0-panel.dtb imx8mp-frdm-boe-wxga-lvds1-panel.dtb
 
 imx8mp-evk-revb4-dtbs := imx8mp-evk.dtb imx8mp-evk-revb4.dtbo
 imx8mp-evk-revb4-rm67191-dtbs := imx8mp-evk-rm67191.dtb imx8mp-evk-revb4.dtbo
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wxga-lvds0-panel.dts b/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wxga-lvds0-panel.dts
new file mode 100644
index 000000000000..210ad3851209
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wxga-lvds0-panel.dts
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx8mp-frdm.dts"
+
+/ {
+	lvds_backlight: lvds_backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm3 0 100000 0>;
+		enable-gpios = <&pcal6416 8 GPIO_ACTIVE_HIGH>;
+		power-supply = <&reg_vdd_12v>;
+		status = "okay";
+
+		brightness-levels = < 0  1  2  3  4  5  6  7  8  9
+				     10 11 12 13 14 15 16 17 18 19
+				     20 21 22 23 24 25 26 27 28 29
+				     30 31 32 33 34 35 36 37 38 39
+				     40 41 42 43 44 45 46 47 48 49
+				     50 51 52 53 54 55 56 57 58 59
+				     60 61 62 63 64 65 66 67 68 69
+				     70 71 72 73 74 75 76 77 78 79
+				     80 81 82 83 84 85 86 87 88 89
+				     90 91 92 93 94 95 96 97 98 99
+				    100>;
+		default-brightness-level = <80>;
+	};
+
+	lvds_panel {
+		compatible = "boe,ev121wxm-n10-1850";
+		backlight = <&lvds_backlight>;
+
+		port {
+			panel_lvds_in: endpoint {
+				remote-endpoint = <&lvds_out>;
+			};
+		};
+	};
+};
+
+&ldb {
+	status = "okay";
+
+	lvds-channel@0 {
+		status = "okay";
+
+		port@1 {
+			reg = <1>;
+
+			lvds_out: endpoint {
+				remote-endpoint = <&panel_lvds_in>;
+			};
+		};
+	};
+};
+
+&ldb_phy {
+	status = "okay";
+};
+
+&i2c2 {
+	exc80h60: touch@2a {
+		compatible = "eeti,exc80h60";
+		reg = <0x2a>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_ctp_int>;
+		interrupt-parent = <&gpio4>;
+		interrupts = <28 IRQ_TYPE_LEVEL_LOW>;
+		reset-gpios = <&pcal6416 10 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+};
+
+&iomuxc {
+	pinctrl_ctp_int: ctp_int_grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_SAI3_RXFS__GPIO4_IO28	0x31e
+		>;
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wxga-lvds1-panel.dts b/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wxga-lvds1-panel.dts
new file mode 100644
index 000000000000..01e5fde89dac
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-frdm-boe-wxga-lvds1-panel.dts
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx8mp-frdm.dts"
+
+/ {
+	lvds_backlight: lvds_backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm2 0 100000 0>;
+		enable-gpios = <&pcal6416 9 GPIO_ACTIVE_HIGH>;
+		power-supply = <&reg_vdd_12v>;
+		status = "okay";
+
+		brightness-levels = < 0  1  2  3  4  5  6  7  8  9
+				     10 11 12 13 14 15 16 17 18 19
+				     20 21 22 23 24 25 26 27 28 29
+				     30 31 32 33 34 35 36 37 38 39
+				     40 41 42 43 44 45 46 47 48 49
+				     50 51 52 53 54 55 56 57 58 59
+				     60 61 62 63 64 65 66 67 68 69
+				     70 71 72 73 74 75 76 77 78 79
+				     80 81 82 83 84 85 86 87 88 89
+				     90 91 92 93 94 95 96 97 98 99
+				    100>;
+		default-brightness-level = <80>;
+	};
+
+	lvds_panel {
+		compatible = "boe,ev121wxm-n10-1850";
+		backlight = <&lvds_backlight>;
+
+		port {
+			panel_lvds_in: endpoint {
+				remote-endpoint = <&lvds_out>;
+			};
+		};
+	};
+};
+
+&ldb {
+	status = "okay";
+
+	lvds-channel@1 {
+		status = "okay";
+
+		port@1 {
+			reg = <1>;
+
+			lvds_out: endpoint {
+				remote-endpoint = <&panel_lvds_in>;
+			};
+		};
+	};
+};
+
+&ldb_phy {
+	status = "okay";
+};
+
+&i2c3 {
+	exc80h60: touch@2a {
+		compatible = "eeti,exc80h60";
+		reg = <0x2a>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_ctp_int>;
+		interrupt-parent = <&gpio4>;
+		interrupts = <28 IRQ_TYPE_LEVEL_LOW>;
+		reset-gpios = <&pcal6416 11 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+};
+
+&iomuxc {
+	pinctrl_ctp_int: ctp_int_grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_SAI3_RXC__GPIO4_IO29	0x31e
+		>;
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp.dtsi b/arch/arm64/boot/dts/freescale/imx8mp.dtsi
index 3e6d8c76d30a..c3c7e36ba5c4 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp.dtsi
@@ -2311,12 +2311,16 @@ ldb_ch0: endpoint {
 		};
 
 		lvds-channel@1 {
+			#address-cells = <1>;
+			#size-cells = <0>;
 			reg = <1>;
 			phys = <&ldb_phy2>;
 			phy-names = "ldb_phy";
 			status = "disabled";
 
-			port {
+			port@0 {
+				reg = <0>;
+
 				ldb_ch1: endpoint {
 					remote-endpoint = <&lcdif2_disp_ldb_ch1>;
 				};
-- 
2.34.1

